test/integration/helmcharts/chartstests.go

```diff
@@ -0,0 +1,211 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package helmcharts
+
+import (
+	"context"
+	"encoding/base64"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gardener/landscaper/pkg/deployer/helm"
+	kutil "github.com/gardener/landscaper/pkg/utils/kubernetes"
+	"github.com/gardener/landscaper/test/framework"
+	"github.com/gardener/landscaper/test/utils"
+	"github.com/gardener/landscaper/test/utils/envtest"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	helmv1alpha1 "github.com/gardener/landscaper/apis/deployer/helm/v1alpha1"
+	"github.com/gardener/landscaper/apis/deployer/helm/v1alpha1/helper"
+)
+
+// RegisterTests registers all tests of the package
+func RegisterTests(f *framework.Framework) {
+	DeployerTests(f)
+}
+
+// DeployerTests tests if the deployers can be deployed into a cluster through their Helm Charts and the Helm Deployer
+func DeployerTests(f *framework.Framework) {
+	dumper := f.Register()
+
+	_ = Describe("DeployerTests", func() {
+
+		It("should deploy the Container deployer through its Helm Chart", func() {
+			ctx := context.Background()
+			defer ctx.Done()
+
+			state, cleanup, err := f.NewState(ctx)
+			utils.ExpectNoError(err)
+			dumper.AddNamespaces(state.Namespace)
+			defer func() {
+				Expect(cleanup(ctx)).ToNot(HaveOccurred())
+			}()
+
+			By("Setting up the Helm deployer first")
+			deployHelmDeployer(ctx, state, f, dumper)
+
+			By("Creating and applying a Container Deployer DeployItem")
+			var (
+				chartDir   = filepath.Join(f.RootPath, "/charts/container-deployer")
+				valuesFile = filepath.Join(chartDir, "values.yaml")
+			)
+
+			const deployerName = "container-deployer"
+
+			target, err := utils.CreateInternalKubernetesTarget(ctx, f.Client, state.Namespace, deployerName, f.RestConfig)
+			utils.ExpectNoError(err)
+			utils.ExpectNoError(state.Create(ctx, f.Client, target))
+
+			di := forgeDeployItem(chartDir, valuesFile, deployerName, target)
+			utils.ExpectNoError(state.Create(ctx, f.Client, di))
+			utils.ExpectNoError(utils.WaitForHelmDeployItemToSucceed(ctx, f.TestLog(), f.Client, kutil.ObjectKey(di.Name, state.Namespace), 2*time.Minute))
+			utils.ExpectNoError(utils.WaitForDeploymentToBeReady(ctx, f.TestLog(), f.Client, kutil.ObjectKey(deployerName, state.Namespace), 2*time.Minute))
```

Comment:
I think we should also test the deletion afterwards.

Reply:
Done.